crates/agglayer-certificate-orchestrator/src/network_task.rs

```diff
@@ -255,459 +287,90 @@ where
-                );
-                match self
-                    .on_proven_certificate(height, certificate, new_state)
-                    .await
-                {
-                    Err(error) => {
-                        error!(
-                            hash = certificate_id.to_string(),
-                            "Error during the certification process of {certificate_id} for \
-                             network {}: {:?}",
-                            self.network_id,
-                            error
-                        );
-                        self.pending_state = None;
-                        self.at_capacity_for_epoch = false;
-                    }
-                    Ok(settled) => {
-                        self.latest_settled = Some(settled);
-                        *next_expected_height += 1;
-                        self.at_capacity_for_epoch = true;
-
-                        debug!(
-                            hash = certificate_id.to_string(),
-                            "Certification process completed for {certificate_id} for network {}",
-                            self.network_id
-                        );
-                    }
-                }
-                Ok(())
-            }
+                header,
+                sender,
+                self.state_store.clone(),
+                self.certifier_client.clone(),
+                self.settlement_client.clone(),
+            )
+            .process(),
+        );
 
-            Err(error) => {
-                warn!(
-                    hash = certificate_id.to_string(),
-                    "Error during certification process of {certificate_id}: {}", error
-                );
-                let error: CertificateStatusError = match error {
-                    CertificationError::CertificateNotFound(network, height) => {
-                        CertificateStatusError::InternalError(format!(
-                            "Certificate not found for network {network} at height {height}"
-                        ))
-                    }
-                    CertificationError::TrustedSequencerNotFound(network) => {
-                        CertificateStatusError::TrustedSequencerNotFound(network)
-                    }
-                    CertificationError::LastPessimisticRootNotFound(network_id) => {
-                        CertificateStatusError::LastPessimisticRootNotFound(network_id)
-                    }
-                    CertificationError::ProofVerificationFailed { source } => {
-                        CertificateStatusError::ProofVerificationFailed(source)
+        loop {
+            tokio::select! {
+                msg = receiver.recv() => match msg {
+                    None => {
+                        error!(height = *next_expected_height, %certificate_id, "Certificate task channel closed");
+                        return Err(Error::InternalError("Certificate task channel closed".into()));
```

Comment:
In case of a certificate already settled (which return `Ok(())` and drop the sender), it means that we end up here, right?